### DISCUSSION TITLE
Mason breaks null-ls config

### DISCUSSION THREAD

**Discussion by Everduin94** [mason-org/mason.nvim#1094] [Q&A]
Hello, I have a null-ls config like this (see below).

Adding `require('mason').setup()`, my config no longer works, i.e the example below no longer formats when calling `lsp.buf.format()`

I tried implementing and following the documentation in `mason-null-ls`. However, then I can only format using `eslint-lsp` and I can't figure out how to change what is doing the formatting. 

I think the idea of this plugin is really cool, but I can't seem to figure out how to integrate it into my existing setup. To reiterate:
- calling mason setup seems to override/nullify behavior in null-ls
- using mason-null-ls seems to use eslint-lsp for formatting and I can't figure out to change that to `prettierd`

My config
```lua
require("null-ls").setup({
    sources = {
      formatting.prettierd,
      formatting.eslint_d,
      code_actions.eslint_d,
      diagnostics.eslint_d,
    }
})

-- Format on save
local augroup = vim.api.nvim_create_augroup("LspFormatting", {})
vim.api.nvim_clear_autocmds({ group = augroup, buffer = bufnr })
vim.api.nvim_create_autocmd("BufWritePre", {
    group = augroup,
    buffer = bufnr,
    callback = function()
      print('firing!')
      vim.lsp.buf.format({
          filter = function(client)
              -- apply whatever logic you want (in this example, we'll only use null-ls)
              return client.name == "null-ls"
          end,
          bufnr = bufnr,
      })
    end,
})
```

**Comment by williamboman** [comment]
Hello! `mason.nvim` does very little during setup, essentially only updating `PATH` and setting up the `:Mason`, `:MasonInstall`, etc. commands. It does not interfere with null-ls in any way. The chances that mason.nvim is the culprit is very low, there was probably some other change that made things stop work the way you expect.

**Reply by Everduin94** [reply]
Dang I have no idea what I'm doing wrong in my config then :( 

In this video I have `mason.setup()` commented out, I run prettier via keymaps & autocommands on a random ts file.

I uncomment `mason.setup()`, restart neovim, try to use the same keymaps & autocommands. Prettier no longer formats. For context I've removed any usage of mason-null-ls or any other mason extension. 

https://user-images.githubusercontent.com/14320878/224996327-7404a9e7-113b-4ceb-8b16-82b5bb6a9a31.mov

**Reply by williamboman** [reply]
Is the `prettierd` version that you have installed the same as the one Mason ends up installing? Do you have `prettier` installed locally in the project?

**Reply by Everduin94** [reply]
Hey thank you for taking the time to reply. I tried both in a project without local prettier (using global prettier_d), and a project at my work (with local prettier). -- same behavior.  I will look further into the versions once I get the chance. Again, thanks!
